lib/AST/DeclContext.cpp

```diff
@@ -1418,6 +1418,9 @@ bool DeclContext::isAsyncContext() const {
 }
 
 SourceLoc swift::extractNearestSourceLoc(const DeclContext *dc) {
+  if (!dc)
+    return SourceLoc();
```

Comment:
This feels like it should be an assert instead and callers should be responsible for action on `nullptr`. It should be possible to add additional checks to `maybeExtractNearestSourceLoc` in `SimpleRequest` to prevent calls to `extractNearestSourceLoc` on invalid inputs.

Reply:
@xedin I added the check in `maybeExtractNearestSourceLoc` for any pointer types instead, let me know what you think

Reply:
Looks good! Please add parenthesis though before merging.

Reply:
@xedin, sorry, parenthesis around what? The constexpr?

Reply:
Sorry, I meant curly braces 🤦 At least for the outer if statement with constexpr, ideally both.